js/src/credentialRequest/types/LdpVcCredentialRequest.js

```diff
@@ -0,0 +1,61 @@
+const Logger = require("../../common/Logger");
+const DownloadFailedException = require("../../exception/DownloadFailedException");
+const CredentialRequest = require("../credentialRequest");
+
+class LdpVcCredentialRequest extends CredentialRequest {
+    constructor(accessToken, issuerMetaData, proof) {
+        super();
+        this.accessToken = accessToken;
+        this.issuerMetaData = issuerMetaData;
+        this.proof = proof;
+    }
+
+    constructRequest() {
+        const request = {
+          requestBody: this.generateRequestBody(),
+          requestHeader: {
+            'Authorization': `Bearer ${this.accessToken}`,
+            'Content-Type': 'application/json',
+          },
+        };
+        return request;
+      }
+
+    generateRequestBody() {
+        try {
+            const requestBody = new CredentialRequestBody(
+              this.issuerMetaData.credentialFormat,
+              new CredentialDefinition(this.issuerMetaData.credentialType),
+              {
+                proof_type: this.proof.proofType,
+                jwt: this.proof.jwt,
+              },
```

Comment:
Are we required to build this `{
                proof_type: this.proof.proofType,
                jwt: this.proof.jwt,
              },` object again? Asking since proof variable itself will have the same structure